egs2/TEMPLATE/asr1/scripts/utils/speechlm_eval/eval_svs.sh

```diff
@@ -0,0 +1,210 @@
+#!/usr/bin/env bash
+
+# Copyright 2024 Jinchuan Tian
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+
+# Set bash to 'debug' mode, it will exit on :
+# -e 'error', -u 'undefined variable', -o ... 'error in pipeline', -x 'print commands',
+set -e
+set -u
+set -o pipefail
+
+log() {
+    local fname=${BASH_SOURCE[1]##*/}
+    echo -e "$(date '+%Y-%m-%dT%H:%M:%S') (${fname}:${BASH_LINENO[0]}:${FUNCNAME[1]}) $*"
+}
+min() {
+  local a b
+  a=$1
+  for b in "$@"; do
+      if [ "${b}" -le "${a}" ]; then
+          a="${b}"
+      fi
+  done
+  echo "${a}"
+}
+SECONDS=0
+
+. ./path.sh
+. ./cmd.sh
+
+stage=1 #TODO(yiwen) try using versa, but some scores like vuv and semitone have not been support in VERSA
+stop_stage=100
+nj=8
+inference_nj=8
+gpu_inference=true
+nbest=1
+
+gen_dir=
+ref_dir=
+key_file=
+eval_mcd=true
+eval_log_F0=true
+eval_semitone_ACC=true
+eval_VUV_res=true
+spk_config=conf/eval_spk.yaml
+mcd_f0_config=conf/eval_mcd_f0.yaml
+
+# wer options
+whisper_tag=large
+whisper_dir=local/whisper
+cleaner=whisper_en
+hyp_cleaner=whisper_en
+
+python=python3
+
+log "$0 $*"
+. utils/parse_options.sh
+
+mkdir -p ${gen_dir}/scoring
+
+if [ ${stage} -le 1 ] && [ ${stop_stage} -ge 1 ]; then
+    log "Using espnet embeded eval functions"
+    _dir="${gen_dir}/scoring"
+
+    if ${eval_mcd}; then
+        mkdir -p "${_dir}/MCD_res"
+        # Objective Evaluation - MCD
+        log "Begin Scoring for MCD metrics, results are written under ${_dir}/MCD_res"
+        ${python} ./pyscripts/utils/evaluate_mcd.py \
+            ${gen_dir}/wav \
+            ${gen_dir}/ref_wav.scp \
+            --outdir "${_dir}/MCD_res"
+    fi
+
+    if ${eval_log_F0}; then
+        mkdir -p "${_dir}/F0_res"
+        # Objective Evaluation - log-F0 RMSE
+        log "Begin Scoring for F0 related metrics, results are written under ${_dir}/F0_res"
+        ${python} pyscripts/utils/evaluate_f0.py \
+            ${gen_dir}/wav \
+            ${gen_dir}/ref_wav.scp \
+            --outdir "${_dir}/F0_res"
+    fi
+
+    if ${eval_semitone_ACC}; then
+        mkdir -p "${_dir}/SEMITONE_res"
+        # Objective Evaluation - semitone ACC
+        log "Begin Scoring for SEMITONE related metrics, results are written under ${_dir}/SEMITONE_res"
+        ${python} pyscripts/utils/evaluate_semitone.py \
+            ${gen_dir}/wav \
+            ${gen_dir}/ref_wav.scp \
+            --outdir "${_dir}/SEMITONE_res"
+    fi
+
+    if ${eval_VUV_res}; then
+        mkdir -p "${_dir}/VUV_res"
+        # Objective Evaluation - VUV error
+        log "Begin Scoring for VUV related metrics, results are written under ${_dir}/VUV_res"
+        ${python} pyscripts/utils/evaluate_vuv.py \
+            ${gen_dir}/wav \
+            ${gen_dir}/ref_wav.scp \
+            --outdir "${_dir}/VUV_res"
+    fi
+
+fi
+
+# if [ ${stage} -le 2 ] && [ ${stop_stage} -ge 2 ]; then
```

Comment:
we can remove the comments below.
@ftshijt  could you review this evaluation script?